tests/integration/high_availability/high_availability_helpers.py

```diff
@@ -231,3 +240,208 @@ async def high_availability_test_setup(ops_test: OpsTest) -> Tuple[str, str]:
     await relate_mysql_and_application(ops_test, mysql_application_name, application_name)
 
     return mysql_application_name, application_name
+
+
+async def send_signal_to_pod_container_process(
+    ops_test: OpsTest, unit_name: str, container_name: str, process: str, signal_code: str
+) -> None:
+    """Send the specified signal to a pod container process.
+
+    Args:
+        ops_test: The ops test framework
+        unit_name: The name of the unit to send signal to
+        container_name: The name of the container to send signal to
+        process: The name of the process to send signal to
+        signal_code: The code of the signal to send
+    """
+    kubernetes.config.load_kube_config()
+
+    pod_name = unit_name.replace("/", "-")
+
+    send_signal_command = f"pkill --signal {signal_code} -f {process}"
+    response = kubernetes.stream.stream(
+        kubernetes.client.api.core_v1_api.CoreV1Api().connect_get_namespaced_pod_exec,
+        pod_name,
+        ops_test.model.info.name,
+        container=container_name,
+        command=send_signal_command.split(),
+        stdin=False,
+        stdout=True,
+        stderr=True,
+        tty=False,
+        _preload_content=False,
+    )
+    response.run_forever(timeout=5)
+
+    assert (
+        response.returncode == 0
+    ), f"Failed to send {signal_code} signal, unit={unit_name}, container={container_name}, process={process}"
+
+
+async def get_process_stat(
+    ops_test: OpsTest, unit_name: str, container_name: str, process: str
+) -> str:
+    """Retrieve the STAT column of a process on a pod container.
+
+    Args:
+        ops_test: The ops test framework
+        unit_name: The name of the unit for the process
+        container_name: The name of the container for the process
+        process: The name of the process to get the STAT for
+    """
+    get_stat_commands = [
+        "ssh",
+        "--container",
+        container_name,
+        unit_name,
+        f"ps aux | grep {process} | grep -v grep | awk '{{print $8}}'",
```

Comment:
nit (eliminate the need for `grep -v`)
```suggestion
        f"ps -eo comm,stat | grep {process} | awk '{{print $2}}'",
```

Reply:
great suggestion! adopted in [521b968](https://github.com/canonical/mysql-k8s-operator/pull/105/commits/521b9688b94e51c9d0a58590f556c17107a8d933)